src/dos/drive_virtual.cpp

```diff
@@ -536,20 +522,21 @@ bool Virtual_Drive::FindNext(DOS_DTA &dta)
 		if (cmp)
 			return true;
 	}
-	while (search_file) {
-		if (pos == search_file->onpos &&
-		    ((attr & DOS_ATTR_DIRECTORY) || !search_file->isdir) &&
-		    WildFileCmp(search_file->name.c_str(), pattern)) {
-			dta.SetResult(search_file->name.c_str(),
-			              search_file->size,
-			              search_file->date,
-			              search_file->time,
-			              (int)(search_file->isdir ? DOS_ATTR_DIRECTORY
-			                                       : DOS_ATTR_ARCHIVE));
-			search_file = search_file->next;
-			return true;
-		}
+	search_file = find_vfile_by_predicate(
+	        search_file, [pos, attr, pattern](vfile_block_t vfile) {
+		        return pos == vfile->onpos &&
+		               ((attr & DOS_ATTR_DIRECTORY) || !vfile->isdir) &&
+		               WildFileCmp(vfile->name.c_str(), pattern);
+	        });
```

Comment:
This is another area where the template predicate plus inline lamda definition creates another reading stumbling block.

Can this be lifted out to a wrapper/helper function with a nice name, similar to the other helpers like `find_vfile_by_name`, `find_vfile_by_name_and_pos`, etc..)?

I realize this new helper function is only used in one place; but the code will still be DRY and made more readable here.

Reply:
+1 on that one; it would improve readability and clarify intent a lot.

Reply:
ok